corehq/apps/ota/decorators.py

```diff
@@ -23,21 +23,18 @@
 def require_mobile_access(fn):
     @wraps(fn)
     def _inner(request, domain, *args, **kwargs):
-        if Domain.get_by_name(domain).restrict_mobile_access:
-            origin_token = request.META.get(ORIGIN_TOKEN_HEADER, None)
-            if origin_token:
-                if _test_token_valid(origin_token):
-                    return fn(request, domain, *args, **kwargs)
-                else:
-                    auth_logger.info(
-                        "Request rejected domain=%s reason=%s request=%s",
-                        domain, "flag:mobile_access_restricted", request.path
-                    )
-                    return HttpResponseForbidden()
-
-            return require_permission(HqPermissions.access_mobile_endpoints)(fn)(request, domain, *args, **kwargs)
-
-        return fn(request, domain, *args, **kwargs)
+        origin_token = request.META.get(ORIGIN_TOKEN_HEADER, None)
```

Comment:
before this get's rolled out I think we should do one more validation on production envs that the roles are as we expect them to be for domains that do not have `restrict_mobile_access` enabled (i.e. we expect all roles to have the permission)

Reply:
That's a good idea, I can do that

Reply:
Commenting here for posterity - there were actually a handful of roles in an unexpected state, likely due to weirdness in the couch-to-sql migration.  Details and resolution here: https://dimagi-dev.atlassian.net/browse/USH-2461